### PR TITLE
update: strucuture revamp, asset handling implementation and overview

### PR DESCRIPTION
**Pull request by masiedu4**
## Activities
- Removed `cartesi-wallet` implementations from main "Development" section
- Added `cartesi-wallet` implementation to tutorials
- One paragraph going into app-chains and security guarantees
- removed all dead links in the GraphQL API reference
- Revamped asset handling section with ABI-encoded diagrams
- Rollups APIs is a standalone section
- `Cartesi Machine` has a dedicated page

## Preview link
https://docs-azure-two.vercel.app/cartesi-rollups/1.3/

### CODE REVIEW

File: cartesi-rollups/tutorials/javascript-wallet.md [hunk 1]
```diff
@@ -44,205 +43,119 @@ Import essential functions and classes from external modules:
...
 
 ```js
-try {
-     const jsonpayload = JSON.parse(payloadStr);
-     console.log("payload is");
-     return router.process(jsonpayload.method, data);
-   } catch (e) {
-     return new Error_out(`failed to process command ${payloadStr} ${e}`);
-   }
+balance_get: (_account: Address) => Balance
```
Comment (gbarros) [inline cartesi-rollups/tutorials/javascript-wallet.md:140]: Idk about having these over here, but I don't feel too strongly about it either.

File: cartesi-rollups/core-concepts/mainnet.md [hunk 2]
```diff
@@ -21,7 +21,10 @@ The codebase may contain some undiscovered vulnerabilities that might put user f
 Developers and users should factor this risk into their decision to use Cartesi Rollups and decide how much of their value to entrust to the system.
 
 ## Security
-So, what steps are we taking to keep user funds safe? **Honeypot** is a dApp designed to prove Cartesi Rollups' security.
+
+The Cartesi Rollups infrastructure is being built based on careful design decisions and a strong code review process that aligns with the mainstream dogma of blockchain technology.
```
[the remark this answers is not in the document]
Comment (masiedu4) [reply cartesi-rollups/core-concepts/mainnet.md:25]: oh, sounds great

File: cartesi-rollups/index.md [hunk 3]
```diff
@@ -23,9 +23,11 @@ Let's delve into the workings of a Cartesi Rollup at a high level.
 ![img](../static/img/v1.3/overview.jpg)
 
 
-At its core, the Cartesi Rollup executes the Cartesi Machine - a robust RISCV deterministic emulator running Linux OS - fueled by ordered inputs and custom application code. Inputs from the data-availability layer are read by the Cartesi Node, where the Cartesi Machine processes them and generates outputs. After the optimistic rollup dispute window passes, these outputs are verifiable and possibly executable on the settlement layer.
+At its core, the Cartesi Rollup executes the Cartesi Machine - a robust RISCV deterministic emulator running Linux OS - fueled by ordered inputs and custom application code. Inputs from the data-availability layer are read by the Cartesi Node, where the Cartesi Machine processes them and generates outputs. After the optimistic rollup dispute window passes, these outputs are verifiable and possibly executable on the settlement layer.
```
Comment (gbarros) [inline cartesi-rollups/index.md:26]: Suggestion of upgrade:
>Inputs sourced from the data-availability layer are read by the Cartesi Node, inside of which the Cartesi Machine processes them and generates outputs.

File: cartesi-rollups/index.md [hunk 4]
```diff
@@ -23,9 +23,11 @@ Let's delve into the workings of a Cartesi Rollup at a high level.
...
+At its core, the Cartesi Rollup executes the Cartesi Machine - a robust RISCV deterministic emulator running Linux OS - fueled by ordered inputs and custom application code. Inputs from the data-availability layer are read by the Cartesi Node, where the Cartesi Machine processes them and generates outputs. After the optimistic rollup dispute window passes, these outputs are verifiable and possibly executable on the settlement layer. 
 
-In its simplest configuration, the Cartesi Rollup framework seamlessly integrates with Ethereum, acting as the sole data availability, consensus, and settlement layer. Input transactions are directed to specific smart contracts, where the dApp code operates to produce outputs. Following the settlement period, these outputs are verified and may also be executed on Ethereum, as required.
+The Cartesi Rollup architecture is application-specific, meaning each dApp within Cartesi's ecosystem runs on its unique, high-performance rollup app chain with its CPU. This prevents resource sharing with other dApps, significantly enhances computational scalability, and preserves the strong security guarantees of the underlying base layer.
```
Comment (gbarros) [inline cartesi-rollups/index.md:28]: "The Cartesi Rollup architecture is application-specific, meaning each dApp within Cartesi's ecosystem runs on its own ~, high-performance~ rollup app chain with its dedicated CPU. This prevents resource sharing with other dApps, significantly enhances computational scalability, and preserves the strong security guarantees of the underlying base layer."

File: cartesi-rollups/index.md [hunk 5]
```diff
@@ -23,9 +23,11 @@ Let's delve into the workings of a Cartesi Rollup at a high level.
...
+
+The Cartesi Rollup framework seamlessly integrates with Ethereum, acting as the sole data availability, consensus, and settlement layer in its most straightforward configuration. Input transactions are directed to specific smart contracts, where the dApp code operates to produce outputs. Following the settlement period, these outputs are verified and may be executed on Ethereum.
```
Comment (gbarros) [inline cartesi-rollups/index.md:30]: I think we should revisit this whole paragraph. Let point a few things I see here:
- Ethereum is not the sole option for deployment, but EVM
- "Input transactions are directed" it seems like an automated thing, but the transaction should be pointing to the right place to begin with.
- "*where* the dApp code operates to produce outputs"; Where is "where" meaning here? The solidity smart contracts?
- Overall, I think there is too little information being transmitted in this whole paragraph. 


Ok, we explained that App Specific means having its own 'chain'. Briefly we mention that follows with some nice advantages. Now we are missing adding a bit of the security guarantees.  Here, at a high level, I think we should explain that as an App Specific *optimistic* rollup, you only need 1 honest actor to ensure the correct result/settlement. Through the _verification game_, disputes guarantees fair results even when there are collusion of every other actor against the honest one.

File: cartesi-rollups/index.md [hunk 6]
```diff
@@ -55,11 +57,12 @@ let finish = { status: "accept" };
 
 ```
 
-To ensure flexibility across different programming languages and stacks, Cartesi dApps are implemented as infinite loops that manage their transaction cycles through HTTP POST requests to the `/finish` endpoint. You can learn more about this abstraction [here](./rollups-apis/backend/introduction.md).
+Cartesi dApps are implemented as infinite loops that manage their transaction cycles through HTTP POST requests to the `/finish` endpoint to ensure flexibility across different programming languages and stacks. You can learn more about this abstraction [here](./rollups-apis/backend/introduction.md).
```
Comment (gbarros) [inline cartesi-rollups/index.md:60]: this is nice!

**Review by gbarros — CHANGES_REQUESTED** [review]
Almost there!

File: cartesi-rollups/index.md (hunk 5 above)
Comment (masiedu4) [reply cartesi-rollups/index.md:30]: Check this out:

```
The Cartesi Rollup framework tailors each DApp with its unique app chain and CPU, enhancing security by linking the dedicated consensus of these rollups directly to the base layer. This setup allows validators —  permissioned or not, to uphold the security standards of the base layer, ensuring that any honest validator can enforce the correct result with its support. The framework aligns closely with the base layer, which serves as the primary platform for data availability, consensus, and settlement. dApp transactions are processed through specific smart contracts to generate verified outputs that, after a settlement period, can be executed on the base layer. 
```

Comment (gbarros) [reply cartesi-rollups/index.md:30]: :100: good to go

Comment (masiedu4) [reply cartesi-rollups/index.md]: All changes pushed. I believe we are good to approve/merge now

**Review by gbarros — APPROVED** [review]

**Pull request merged**